server/src/main/java/org/elasticsearch/transport/ConnectionProfile.java

```diff
@@ -134,6 +134,7 @@ public static ConnectionProfile buildSingleChannelProfile(
     private final TimeValue pingInterval;
     private final Compression.Enabled compressionEnabled;
     private final Compression.Scheme compressionScheme;
+    private final String transportProfile;
```

Comment:
This new field is the key decision of this PR. We need pass the transport profile information all the way to `Netty4Transport`. Instead of having cascading method signature changes, I think it is better to add a new field to the existing `ConnectionProfile` parameter. The purpose of this class also overall fees suitable for housing this new field.

There are still a few method signature changes because `ConnectionProfile` does not get passed down to all necessary level. When initiating TCP channel, all we had is a `DiscoveryNode`. So these methods are refactored to take one more parameter. But the changeset is much smaller.

Reply:
Can we run this decision past the distributed area ? It couples Connection and Transport profiles with a 1:1 cardinality.  I think that is fine today but not sure if that would run against any future plans/designs.

Reply:
This seems fine to me. We already use the `ConnectionProfile` for other settings along these lines. Seems fine to expose it to the lower transport.